cart/infrastructure/InMemoryBehaviour.go

```diff
@@ -354,6 +355,24 @@ func (cob *InMemoryBehaviour) ApplyVoucher(ctx context.Context, cart *domaincart
 	return cob.resetPaymentSelectionIfInvalid(ctx, cart)
 }
 
+// RemoveVoucher removes a voucher from the cart
+func (cob *InMemoryBehaviour) RemoveVoucher(ctx context.Context, cart *domaincart.Cart, couponCode string) (*domaincart.Cart, domaincart.DeferEvents, error) {
+	for i, coupon := range cart.AppliedCouponCodes {
+		if coupon.Code == couponCode {
+			cart.AppliedCouponCodes[i] = cart.AppliedCouponCodes[len(cart.AppliedCouponCodes)-1]
+			cart.AppliedCouponCodes[len(cart.AppliedCouponCodes)-1] = domaincart.CouponCode{}
+			cart.AppliedCouponCodes = cart.AppliedCouponCodes[:len(cart.AppliedCouponCodes)-1]
+		}
```

Comment:
We assume that only one coupon code matches, right? In that case you could insert a break after the removal of the element, as there is no need to iterate through the codes further

Reply:
Good point, I added the break.